### PR TITLE
[PM-15970] Allow assigning collections if user has correct permissions

### PR DESCRIPTION
**Pull request by SaintPatrck**
## 🎟️ Tracking

[PM-15970](https://bitwarden.atlassian.net/browse/PM-15970)
[PM-15969](https://bitwarden.atlassian.net/browse/PM-15969)

## 📔 Objective

Allows assigning items to collections if the user has manage permissions and is not restricted from viewing or editing password in the collection. This fixes an issue where collection assignment would be blocked if the user could view the item and its passwords.

## 📸 Screenshots

Coming soon!

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-15970]: https://bitwarden.atlassian.net/browse/PM-15970?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PM-15969]: https://bitwarden.atlassian.net/browse/PM-15969?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/util/model/CollectionPermission.kt
Comment (SaintPatrck) [inline app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/util/model/CollectionPermission.kt:1]: @dseverns-livefront @david-livefront This seemed like the best package for the enum, but I'm open to suggestion if there's a more appropriate place.

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/util/model/CollectionPermission.kt:1]: makes sense to me, another thought I had was: if we are not using this outside the extensions it could live in that file and maybe annotate with `@VisibleForTesting` so if we were to try and use it outside there or the test we get a lint warning, but that may be more philosophical. 

TLDR: Placement makes sense to me.

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/util/CollectionViewExtensions.kt [hunk 1]
```diff
@@ -110,15 +111,43 @@ fun List<CollectionView>?.hasDeletePermissionInAtLeastOneCollection(
...
+/**
+ * Determines the user's permission level for a given [CollectionView].
+ */
+val CollectionView.permission: CollectionPermission
+    get() = when {
+        manage -> CollectionPermission.MANAGE
+        readOnly && hidePasswords -> CollectionPermission.VIEW_EXCEPT_PASSWORDS
+        readOnly -> CollectionPermission.VIEW
+        !readOnly && hidePasswords -> CollectionPermission.EDIT_EXCEPT_PASSWORD
+        // !readOnly is the only other possible condition, which resolves to EDIT permission
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/util/CollectionViewExtensions.kt:151]: 👍

**Review by dseverns-livefront — APPROVED** [review]
Just a small typo and a visibility _option_. But otherwise LGTM!

**Review by dseverns-livefront — APPROVED** [review]
